.fides/14acee6f5459_translation_data.py

```diff
@@ -0,0 +1,312 @@
+            components.append("privacy_center")
+
+        return components
+
+    # Convert displayed_in_* fields to a list of components for consistency
+    if not df.empty:
+        df["component"] = df.apply(transform_components, axis=1)
+    else:
+        df["component"] = pd.Series(dtype=str)
+
+    df = df.drop(["displayed_in_overlay", "displayed_in_privacy_center"], axis=1)
+
+    # Expand to Notice x Region x Component
+    df_component_expanded = df.explode("component")
+
+    df_region_expanded = (
+        df_component_expanded.explode("regions")
+        .drop_duplicates()
+        .reset_index(drop=True)
+    )
+    df_region_expanded = df_region_expanded.rename(columns={"regions": "region"})
+
+    # Join Notice data with Experience data so we have Notice x Region x Component x Experience Config ID
+    merged = pd.merge(exp_df, df_region_expanded, on=["region", "component"])
+
+    def aggregate_list_fields(list_field):
+        """Aggregate notices as a comma-separated string because we still need to do another group by
+        and we can't do that on a list"""
+        sorted_list = sorted(list(set(list_field)))
+        return ",".join(sorted_list)
+
+    # Collapse to |Notice| x Region x Component x Config
+    grouped_df = (
+        merged.groupby(["component", "region", "experience_config_id"])["id"]
+        .agg(aggregate_list_fields)
+        .reset_index()
+    )
+
+    # Collapse to |Notice| x |Region| x Component x Config
+    next_group = (
+        grouped_df.groupby(["id", "component", "experience_config_id"])["region"]
+        .agg(aggregate_list_fields)
+        .reset_index()
+    )
+
+    def convert_csstring_to_list_field(string_field):
+        return string_field.str.split(",")
+
+    next_group[["id", "region"]] = next_group[["id", "region"]].apply(
+        lambda x: convert_csstring_to_list_field(x)
+    )
+    next_group["new_experience_config_needed"] = False
+
+    next_group.loc[
+        next_group.experience_config_id.duplicated(), "new_experience_config_needed"
+    ] = True
+
+    def get_experience_config_id_after(row):
+        if not row["new_experience_config_needed"]:
+            return row["experience_config_id"]
+
+        if (
+            set(row["region"]) == {"us_ca", "us_co", "us_ct", "us_ia", "us_ut", "us_va"}
+            and row["component"] == "overlay"
+            and row["experience_config_id"] == "pri-7ae3-f06b-4096-970f-0bbbdef-over"
+        ):
+            return "pri-4b6e-1165-4e12-8125-18f2db8-over"
+
+        return generate_record_id("pri")
+
+    if not next_group.empty:
+        next_group["experience_config_id_after"] = next_group.apply(
+            get_experience_config_id_after, axis=1
+        )
+    else:
+        next_group["experience_config_id_after"] = pd.Series(dtype=str)
+    return next_group
+
+
+def upgrade():
+    return
+    bind = op.get_bind()
+
+    experience_config_df: DataFrame = determine_needed_experience_configs(bind)
```

Comment:
Main point: 

- We need to create a unique set of Experience Configs to surface the same number of notices that we were surfacing before.
- Source of the new Experience Config
- One ExperienceConfig will stay?